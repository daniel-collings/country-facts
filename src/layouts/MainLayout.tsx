import { ReactNode } from 'react'
import { useTheme } from '@/hooks/useTheme.tsx'

export default function MainLayout({ children }: { children: ReactNode }) {
  const { theme } = useTheme()
  return (
    <div
      data-theme={theme}
      className="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    >
      <main className="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8">{children}</main>
    </div>
  )
}
