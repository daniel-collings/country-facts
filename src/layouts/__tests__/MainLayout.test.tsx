import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import MainLayout from '@/layouts/MainLayout.tsx'

describe('MainLayout Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <MainLayout>
        <h1>Hello World</h1>
      </MainLayout>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
