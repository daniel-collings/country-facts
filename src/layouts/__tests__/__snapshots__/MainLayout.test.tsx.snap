// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout Component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    >
      <div
        class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
        data-theme="light"
      >
        <main
          class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
        >
          <h1>
            Hello World
          </h1>
        </main>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
